lib/puppet/provider/common.rb

```diff
@@ -121,3 +121,8 @@ def extract_resource_name(class_name)
   shift_prefix = Regexp.last_match.nil? ? 2 : 1
   class_name.split('::')[2].split('_').drop(shift_prefix).collect(&:capitalize).join
 end
+
+def create_image_streamer_client
```

Comment:
I created this method on common.rb to avoid code duplication on the ImageStreamerResource.

If I move it to the ImageStreamerResource, it will need two methods doing the same stuff: ```create_image_streamer_client``` and ```self.create_image_streamer_client```.

If I just move it, as-is now, it will throw exceptions when I call the `self.instances` method:

```ruby
   Failure/Error: self.create_image_streamer_client
     
     NoMethodError:
       undefined method `create_image_streamer_client' for Puppet::Type::Image_streamer_plan_script::ProviderImage_streamer:Class
``` 

:disappointed: 

Do you have any idea?